### DISCUSSION TITLE
How to split documents from one file by --- into an array?

### DISCUSSION THREAD

**Discussion by shi-ron** [mikefarah/yq#1791] [Q&A]
Hi, if we have data.yml like:
```yaml
A: 1
B: 2
---
C: 3
D: 4
```
and we wish to run a command like:
```bash
readarray DATA < <(yq -s '---' data.yml)
```
The array `$DATA` shall has two elements.
```bash
echo "${DATA[0]}"
```
should be
```
A: 1
B: 2
```
and
```
echo "${DATA[1]}"
```
should be
```
C: 3
D: 4
```
How can I do this? Thank you for your help!

**Comment by mikefarah** [comment]
I don't know how readarray works - but to get all the documents into a single yaml array you can do this:

```
yq ea '[.]' data.yaml
```


```
- A: 1
  B: 2
- C: 3
  D: 4
```

**Comment by mikefarah** [comment]
Forgot to explain:
- `ea` is shorthand for `eval-all` which means all the yaml documents are read into memory before running the expression. By default, the expression is run against a single document at a time.
- `[.]` puts everything in the current context `.` into an array. Which in in this case, will put all the documents.

**Comment by shi-ron** [comment]
I see the `ea` has merged the documents. But I want to iterate them in a loop.

**Comment by mikefarah** [comment]
Now that it's in an array - you can follow the solution here: https://stackoverflow.com/questions/62898925/using-yq-in-for-loop-bash/70162342#70162342

**Comment by shi-ron** [comment]
Ok, it works! Thank you.
